pkg/cmd/util/clientcmd/negotiate.go

```diff
@@ -24,18 +27,22 @@ func negotiateVersion(client *kclient.Client, config *restclient.Config, request
 		}
 	}
 
-	// Get server versions
-	serverGVs, err := serverAPIVersions(client, "/oapi")
-	if err != nil {
-		return nil, err
-	}
-
 	// Determine our preferred version
 	preferredGV := copyGroupVersion(requestedGV)
 	if preferredGV == nil {
 		preferredGV = copyGroupVersion(config.GroupVersion)
 	}
 
+	// Get server versions
+	serverGVs, err := serverAPIVersions(client, "/oapi")
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
did you figure out why 404s were not showing up as NotFound api errors? any chance of that happening here?